python/lsst/daf/butler/core/location.py

```diff
@@ -87,7 +97,7 @@ def posix2os(posix):
 
     # Trailing "/" is stripped so we need to add back an empty path
     # for consistency
-    if posix.endswith(posixpath.sep):
+    if str(posix).endswith(posixpath.sep):
```

Comment:
Given that this code worked until now, is it necessary that `posix` be allowed to be a `PurePath`? If so, then the documentation needs to be updated.